src/main/java/net/shadowmage/ancientwarfare/npc/entity/faction/NpcFactionSpellcasterWizardry.java

```diff
@@ -0,0 +1,182 @@
+ * which cannot have any references to EBWizardry objects */
+public class NpcFactionSpellcasterWizardry extends NpcFaction implements ISpellCaster {
+
+	private EntityAIAttackSpell<NpcFactionSpellcasterWizardry> spellCastingAI = new EntityAIAttackSpell<>(this, 0.75D, 16.0F, 32, 30);
+
+	// Field implementations
+	private List<Spell> spells = new ArrayList<Spell>(4);
+	private Spell continuousSpell;
+	private int spellCounter;
+
+	@SuppressWarnings("unused")
+	public NpcFactionSpellcasterWizardry(World world) {
+		super(world);
+		addAI();
+		tasks.addTask(3, spellCastingAI);
+	}
+
+	@SuppressWarnings("unused")
+	public NpcFactionSpellcasterWizardry(World world, String factionName) {
+		super(world, factionName);
+		addAI();
+		tasks.addTask(3, spellCastingAI);
+	}
+
+	@SuppressWarnings("Duplicates")
+	private void addAI() {
+		tasks.addTask(0, new EntityAISwimming(this));
+		tasks.addTask(0, new EntityAIRestrictOpenDoor(this));
+		tasks.addTask(0, new NpcAIDoor(this, true));
+		tasks.addTask(1, new NpcAIFollowPlayer(this));
+		tasks.addTask(2, new NpcAIMoveHome(this, 50.f, 5.f, 30.f, 5.f));
+		tasks.addTask(2, new NpcAIFactionArcherStayAtHome(this));
+
+		tasks.addTask(101, new EntityAIWatchClosest2(this, EntityPlayer.class, 3.0F, 1.0F));
+		tasks.addTask(102, new NpcAIWander(this));
+		tasks.addTask(103, new NpcAIWatchClosest(this, EntityLiving.class, 8.0F));
+
+		targetTasks.addTask(1, new NpcAIHurt(this));
+		targetTasks.addTask(2, new NpcAIAttackNearest(this, this::isHostileTowards));
+	}
+
+	public void setSpells(List<Spell> spells) {
+		this.spells = spells;
+	}
+
+	@Override
+	public List<Spell> getSpells() {
+		return spells;
+	}
+
+	@Override
+	public void setSpellCounter(int count) {
+		spellCounter = count;
+	}
+
+	@Override
+	public int getSpellCounter() {
+		return spellCounter;
+	}
+
+	@Override
+	public SpellModifiers getModifiers() {
+		SpellModifiers spellModifiers = new SpellModifiers();
+		spellModifiers.set("potency", 1.5f, true);
+		return spellModifiers;
+	}
+
+	@Override
+	public void setContinuousSpell(Spell spell) {
+		continuousSpell = spell;
+	}
+
+	@Override
+	public Spell getContinuousSpell() { return continuousSpell; }
+
+	@Override
+	public int getAimingError(EnumDifficulty difficulty) { // for spells
+		switch (difficulty) {
+			case EASY:
+				return 7;
+			case NORMAL:
+				return 4;
+			case HARD:
+				return 1;
+			default:
+				return 7; // Peaceful counts as easy
+		}
+	}
+
+	@Override
+	public boolean hasAltGui() {
+		return true;
+	}
+
+	@Override
+	public void openAltGui(EntityPlayer player) {
+		NetworkHandler.INSTANCE.openGui(player, NetworkHandler.GUI_NPC_FACTION_SPELLCASTER_WIZARDRY, getEntityId(), 0, 0);
+	}
+
+	@Override
+	public void writeEntityToNBT(NBTTagCompound nbt) {
+		super.writeEntityToNBT(nbt);
+		nbt.setTag("spells", NBTExtras.listToNBT(getSpells(), spell -> new NBTTagString(spell.getRegistryName().toString())));
```

Comment:
Use NBTHelper.getTagList instead of using EB's NBTExtras.listToNBT. The reason is that we should really only have dependency on external stuff where absolutely necessary and not where we have our own code that we can use or where we can have something on our side to use. That makes it easier to maintain going forward.